internal/bminventory/inventory.go

```diff
@@ -3072,17 +3062,20 @@ func (b *bareMetalInventory) getHost(ctx context.Context, clusterId string, host
 	return &host, nil
 }
 
-func (b *bareMetalInventory) getCluster(ctx context.Context, clusterID string, returnHosts bool) (*common.Cluster, error) {
+type returnHosts bool
+
+func (b *bareMetalInventory) getCluster(ctx context.Context, clusterID string, flags ...interface{}) (*common.Cluster, error) {
 	log := logutil.FromContext(ctx, b.log)
 	var cluster common.Cluster
-	var db *gorm.DB
-	if returnHosts {
-		db = b.db.Preload("Hosts")
-	} else {
-		db = b.db
+	db := b.db.Unscoped()
```

Comment:
Why unscoped here?  All APIs that previously returned 404 will now work on deleted clusters.  It should be an option (returnDeleted, default=false).